test/e2e/metering_manual_install_test.go

```diff
@@ -107,6 +111,105 @@ func testManualMeteringInstall(
 	assert.NoError(t, err, "capturing logs and uninstalling metering should produce no error")
 }
 
+func createNFSProvisioner(ctx *deployframework.DeployerCtx) error {
+	//Waiting for Metering pods
+	ctx.TargetPodsCount = nonHDFSTargetPodCount
+
+	_, err := createTestingNamespace(ctx.Client, nfsTestingNamespace)
+	if err != nil {
+		return err
+	}
+
+	files := map[string]string{
+		"pod":          "server.yaml",
+		"service":      "service.yaml",
+		"storageClass": "storageclass.yaml",
+	}
+	server := &corev1.Pod{}
+	service := &corev1.Service{}
+	storageClass := &storagev1beta1.StorageClass{}
+	for name, file := range files {
+		absFile := filepath.Join(repoPath, testNFSManifestPath, file)
+		switch name {
+		case "pod":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, server); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Pods(nfsTestingNamespace).Create(context.TODO(), server, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "service":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, service); err != nil {
+				return err
+			}
+			ctx.Client.CoreV1().Services(nfsTestingNamespace).Create(context.TODO(), service, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		case "storageClass":
+			if err := deploy.DecodeYAMLManifestToObject(absFile, storageClass); err != nil {
+				return err
+			}
+			ctx.Client.StorageV1beta1().StorageClasses().Create(context.TODO(), storageClass, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+	}
+	// wait and Poll , get service
+	svc, err := ctx.Client.CoreV1().Services(nfsTestingNamespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+	if apierrors.IsNotFound(err) && svc.Spec.ClusterIP == "" {
+
+		err = wait.Poll(3*time.Second, 5*time.Minute, func() (bool, error) {
+
+			svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
```

Comment:
You missing the error handling for this client-go API call. If you're creating the service resource for the first time, you're risking an error down the line when attempting to grab the `svc.Spec.ClusterIP` value.

I would suggest doing something like the following:
```go
	svc, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
	if apierrors.IsNotFound(err) {
		return false, err
	}
	if err != nil {
		return false, nil
	}

	return svc.Spec.ClusterIP != "", nil
```

Reply:
This makes sense, but this code above is a bit off from what is in there.
The biggest difference is, that I have a long closure going on, which makes it difficult to see where the error handling should nest ...
I have a guess but would you mind showing this with lines 161 -172